### PR TITLE
[BEAM-3983] Add utils for converting to BigQuery types

### PR DESCRIPTION
**Pull request by apilloud**
Add a class for converting Beam Schema to BigQuery types.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [X] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [X] Write a pull request description that is detailed enough to understand:
   - [X] What the pull request does
   - [X] Why it does it
   - [X] How it does it
   - [X] Why this approach
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by apilloud** [comment]
@akedin This is missing gradle support still, but otherwise adds support for automatically writing SQL to BigQuery.

File: pom.xml [hunk 1]
```diff
@@ -110,6 +110,7 @@
     <args4j.version>2.33</args4j.version>
     <avro.version>1.8.2</avro.version>
     <bigquery.version>v2-rev374-1.22.0</bigquery.version>
+    <bigquery.cloud.version>1.22.0</bigquery.cloud.version>
```
[the remark this answers is not in the document]
Comment (apilloud) [reply pom.xml]: This is the google client version. It seems it is already defined here, so I dropped the redundant definition.

**Comment by apilloud** [comment]
R: @kennknowles The first two commits here are API changes in #4991. The second two support bigquery.

**Comment by apilloud** [comment]
run python precommit

**Comment by apilloud** [comment]
I've updated this to use Beam Schemas and moved the common stuff out of SQL.

@reuvenlax You should make sure I'm not doing anything crazy in 4cfa16b050c80d73199ff790d0327d854a2af608

**Comment by apilloud** [comment]
run java precommit

**Comment by kennknowles** [comment]
I've only had a minute to skim this - is it expected that we can now write an integration test using SQL syntax? I didn't see one so I just thought I'd ask. Follow-up: are things designed such that we could use a fake BigQuery so that it can be a unit test?

**Comment by kennknowles** [comment]
run java precommit

**Comment by kennknowles** [comment]
Incidentally, I don't know if Java precommit is timing out due to this PR or not. The timeout seems to be 90 minutes. That should be enough, but OTOH we don't really benefit from having tight timeouts so I'd support bumping it if there's any chance at all of a successful build exceeding the timeout.

**Comment by apilloud** [comment]
There are unit tests of all the pieces but no integration tests. Talking to BigQuery is handled by `BigQueryIO`, which isn't changed here. If there was a mock that can be plugged into BigQueryIO, we could test end to end without BigQuery.

**Comment by apilloud** [comment]
As requested on the dev list, I've dropped this PR down to a single change. It now only covers converting from beam schema to bigquery types. There is no SQL change, that will come in a followup PR.

@reuvenlax can you please take a look?

**Comment by apilloud** [comment]
run java precomit

**Comment by apilloud** [comment]
run java precommit

**Comment by apilloud** [comment]
run python precommit

**Comment by apilloud** [comment]
run java precommit

**Comment by apilloud** [comment]
run java precommit

**Comment by apilloud** [comment]
run java precommit

**Comment by apilloud** [comment]
run java precommit

**Comment by apilloud** [comment]
run java precommit

**Comment by apilloud** [comment]
run java precommit

**Comment by apilloud** [comment]
run java precommit

**Comment by apilloud** [comment]
run java precommit

**Comment by apilloud** [comment]
This one was actually broken after rebase because `getComponentType` was renamed to `getCollectionElementType`. Rebase and fixed.

**Comment by apilloud** [comment]
Now it's green.

**Comment by kennknowles** [comment]
LGTM

**Pull request merged**